Minerals.gd

```diff
@@ -0,0 +1,75 @@
+extends Spatial
+
+
+# Declare member variables here. Examples:
+enum {
+	NODE,
+	PASSED_TIME,
+	RESPAWN_TIME,
+	STATE
+}
+
+var scene = null
+var mineral = null
+var listBroken = [] # [[mineral, 0, 30, false]] in it 
+var thread
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	scene = get_tree().get_root()
+	loadStateMinerals()
+	thread = Thread.new()
+	thread.start(self, "respawnMinerals", listBroken)
+
+# warning-ignore:shadowed_variable
```

Comment:
Please don't disable warnings, if you have such warnings consider fixing them instead of hiding them, because if somehow one of this warning was important you would have missed it, and it would be more difficult to debug